pkg/distributors/contracts/DistributionScheduler.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/SafeERC20.sol";
+
+import "./interfaces/IMultiDistributor.sol";
+
+// solhint-disable not-rely-on-time
+
+/**
+ * Scheduler for MultiDistributor contract
+ */
+contract DistributionScheduler {
+    using SafeERC20 for IERC20;
+
+    IMultiDistributor private immutable _multiDistributor;
+
+    constructor(IMultiDistributor multiDistributor) {
+        _multiDistributor = multiDistributor;
+    }
+
+    enum DistributionStatus { UNINITIALIZED, PENDING, STARTED }
+
+    struct ScheduledDistribution {
+        bytes32 distributionId;
+        IERC20 stakingToken;
+        IERC20 distributionToken;
+        uint256 startTime;
+        address owner;
+        uint256 amount;
+        DistributionStatus status;
+    }
+
+    event DistributionScheduled(
+        bytes32 scheduleId,
+        address indexed owner,
+        IERC20 indexed stakingToken,
+        IERC20 indexed distributionToken,
+        uint256 startTime,
+        uint256 amount
+    );
+    event DistributionStarted(
+        bytes32 scheduleId,
+        address indexed owner,
+        IERC20 indexed stakingToken,
+        IERC20 indexed distributionToken,
+        uint256 startTime,
+        uint256 amount
+    );
+
+    mapping(bytes32 => ScheduledDistribution) private _scheduledDistributions;
+
+    function getScheduledDistributionInfo(bytes32 scheduleId)
+        external
+        view
+        returns (ScheduledDistribution memory reward)
+    {
+        return _scheduledDistributions[scheduleId];
+    }
+
+    function startDistributions(bytes32[] calldata scheduleIds) external {
```

Comment:
Nitpicky, but I'd place this after `schedule`

Reply:
Also let's return the scheduled id here.

Reply:
Why would we return the `sheduleIds`?

Reply:
Sorry, I confused this function with `schedule` and thought it was missing a return value. `schedule` does return the scheduled id, so we're fine.